### PR TITLE
Clean include  BV/BVH/math/mesh_loader

### PR DESCRIPTION
**Pull request by LucileRemigy**

### CODE REVIEW

File: include/hpp/fcl/BV/RSS.h [hunk 1]
```diff
@@ -142,11 +142,7 @@ class RSS
 
 };
 
-
-/// @brief Translate the RSS bv
-RSS translate(const RSS& bv, const Vec3f& t);
-
-/// @brief distance between two RSS bounding volumes
+/// @brief distance between two RSS boundcd tting volumes
```
Comment (jmirabel) [inline include/hpp/fcl/BV/RSS.h]: I think you didn't want to modify the doc here.

File: include/hpp/fcl/math/transform.h [hunk 2]
```diff
@@ -272,13 +266,8 @@ class Transform3f
 /// @brief inverse the transform
 Transform3f inverse(const Transform3f& tf);
```
Comment (jmirabel) [inline include/hpp/fcl/math/transform.h]: This function can be replaced by Transform3f::inverse everywhere it is used.

File: include/hpp/fcl/mesh_loader/loader.h [hunk 3]
```diff
@@ -107,7 +87,7 @@ namespace fcl {
       };
       typedef std::map <Key, CollisionGeometryPtr_t> Cache_t;
 
-      const Cache_t cache () const { return cache_; }
```
Comment (jmirabel) [inline include/hpp/fcl/mesh_loader/loader.h]: I would keep this.

File: include/hpp/fcl/mesh_loader/loader.h [hunk 4]
```diff
@@ -53,16 +53,6 @@ namespace fcl {
...
-      /// \param bvType ignored
-      /// \deprecated Use MeshLoader::load(const std::string&, const Vec3f&)
-      CollisionGeometryPtr_t load (const std::string& filename,
-          const Vec3f& scale,
-          const NODE_TYPE& bvType) HPP_FCL_DEPRECATED
-      {
-        (void) bvType;
-        return load (filename, scale);
-      }
-
```
Comment (jmirabel) [inline include/hpp/fcl/mesh_loader/loader.h]: We must wait a bit before removing this function. It is still used in some dependencies.

File: include/hpp/fcl/mesh_loader/loader.h [hunk 5]
```diff
@@ -83,16 +73,6 @@ namespace fcl {
...
 
-      /// \param bvType ignored
-      /// \deprecated Use MeshLoader::load(const std::string&, const Vec3f&)
-      CollisionGeometryPtr_t load (const std::string& filename,
-          const Vec3f& scale,
-          const NODE_TYPE& bvType) HPP_FCL_DEPRECATED
-      {
-        (void) bvType;
-        return load(filename, scale);
-      }
```
Comment (jmirabel) [inline include/hpp/fcl/mesh_loader/loader.h]: This one can be removed.

File: include/hpp/fcl/BV/RSS.h (hunk 1 above)
Comment (nim65s) [inline include/hpp/fcl/BV/RSS.h]: looks like a typo :)

**Comment by nim65s** [comment]
Also, could you squash your commits ? You are making some changes in the first that are reverted in the second, which would mess the history.

File: include/hpp/fcl/BV/AABB.h [hunk 6]
```diff
@@ -186,24 +169,12 @@ class AABB
     return max_[2] - min_[2];
   }
 
-  /// @brief Volume of the AABB
-  inline FCL_REAL volume() const
-  {
-    return width() * height() * depth();
-  }  
-
```
Comment (jmirabel) [inline include/hpp/fcl/BV/AABB.h]: In order to homogenize the API of all bounding volumes, I would keep this.

File: include/hpp/fcl/BV/OBBRSS.h [hunk 7]
```diff
@@ -74,18 +74,6 @@ class OBBRSS
...
-  bool overlap(const OBBRSS& other, OBBRSS& /*overlap_part*/) const
-  {
-    return overlap(other);
-  }
-
-  /// @brief Check whether the OBBRSS contains a point
-  inline bool contain(const Vec3f& p) const
-  {
-    return obb.contain(p);
-  }
```
Comment (jmirabel) [inline include/hpp/fcl/BV/OBBRSS.h]: In order to homogenize the API of all bounding volumes, I would keep method *contain*.

File: include/hpp/fcl/BVH/BVH_model.h [hunk 8]
```diff
@@ -123,9 +147,6 @@ class BVHModel : public CollisionGeometry,
     return num_bvs;
   }
 
-  /// @brief Get the object type: it is a BVH
-  OBJECT_TYPE getObjectType() const { return OT_BVH; }
-
```
Comment (jmirabel) [inline include/hpp/fcl/BVH/BVH_model.h]: This shall not be changed.

File: include/hpp/fcl/BVH/BVH_model.h [hunk 9]
```diff
@@ -186,90 +205,6 @@ class BVHModel : public CollisionGeometry,
...
-    for(int i = 0; i < num_tris; ++i)
-    {
-      const Triangle& tri = tri_indices[i];
-      const Vec3f& v1 = vertices[tri[0]];
-      const Vec3f& v2 = vertices[tri[1]];
-      const Vec3f& v3 = vertices[tri[2]];
-      Matrix3f A; A << v1.transpose(), v2.transpose(), v3.transpose();
-      C += A.derived().transpose() * C_canonical * A * (v1.cross(v2)).dot(v3);
-    }
```
Comment (jmirabel) [inline include/hpp/fcl/BVH/BVH_model.h]: The 4 functions above should be kept.

File: include/hpp/fcl/math/transform.h (hunk 2 above)
Comment (LucileRemigy) [reply include/hpp/fcl/math/transform.h]: which function can be replaced?

Comment (jmirabel) [reply include/hpp/fcl/math/transform.h]: `Transform3f Minv = inverse (M);` can be replaced by `Transform3f Minv = M.inverse ();

File: include/hpp/fcl/math/types.h [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+#include <hpp/fcl/data_types.h>
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <hpp/fcl/math/tools.h>
+
+#include <cmath>
+#include <iostream>
+#include <limits>
```
Comment (jmirabel) [inline include/hpp/fcl/math/types.h]: Can you remove those lines:
```cpp
#include <hpp/fcl/math/tools.h>

#include <cmath>
#include <iostream>
#include <limits>
```

File: include/hpp/fcl/math/types.h [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+namespace hpp
+{
+namespace fcl
+{
+  typedef Eigen::Matrix<FCL_REAL, 3, 1> Vec3f;
+}
```
Comment (jmirabel) [inline include/hpp/fcl/math/types.h]: Move this typedef with the typedef Matrix3f below.

File: include/hpp/fcl/math/types.h (hunk 10 above)
Comment (LucileRemigy) [reply include/hpp/fcl/math/types.h]: When I remove this line "#include <hpp/fcl/math/tools.h>", the project does not compile. I think, we can't remove it.

Comment (jmirabel) [reply include/hpp/fcl/math/types.h]: You may need to include it in other cc or hh files.

**Comment by jmirabel** [comment]
@LucileRemigy I wait for travis to complete. The previous failure is, I believe, a problem of the unit test itself. Could you wait for this PR to be merged before pushing new commits ?

@nim65s any other comments ?

**Comment by nim65s** [comment]
I tried to run it on [Gitlab CI](https://gepgitlab.laas.fr/gsaurel/hpp-fcl/pipelines/5433), and it seems that  one test is failing in 16.04 on Debug mode:

```
 1/15 Test  #1: test_fcl_math ....................   Passed    0.13 sec
      Start  5: test_fcl_geometric_shapes
 2/15 Test  #5: test_fcl_geometric_shapes ........***Failed    1.43 sec
Running 42 test cases...
p2 in cylinder = (-0.0295784 0.00332625  0.0512009)
p1 in box = (  0.40428 -0.300945 0.0135652)
p2 in cylinder = (   0.433724 -0.00562366    0.443942)
p1 in box = ( -1.38168 -0.119498 -0.215351)
test_fcl_geometric_shapes: …/include/hpp/fcl/narrowphase/narrowphase.h:198: bool hpp::fcl::GJKSolver::shapeDistance(const S1&, const hpp::fcl::Transform3f&, const S2&, const hpp::fcl::Transform3f&, hpp::fcl::FCL_REAL&, hpp::fcl::Vec3f&, hpp::fcl::Vec3f&, hpp::fcl::Vec3f&) const [with S1 = hpp::fcl::Cylinder; S2 = hpp::fcl::Box; hpp::fcl::FCL_REAL = double; hpp::fcl::Vec3f = Eigen::Matrix<double, 3, 1>]: Assertion `distance * distance < sqrt (eps)' failed.
unknown location(0): fatal error in "shapeIntersection_cylinderbox": signal: SIGABRT (application abort requested)
…/test/test_fcl_geometric_shapes.cpp(327): last checkpoint

*** 1 failure detected in test suite "FCL_GEOMETRIC_SHAPES"

```

**Comment by jmirabel** [comment]
Yes, but this bug is not related to those changes. A fix may come soon.

**Pull request merged**